runtime/drivers/duckdb/connectors.go

```diff
@@ -197,6 +316,10 @@ func generateReadCsvStatement(paths []string, properties map[string]any) (string
 	if _, sampleSizeDefined := ingestionProps["sample_size"]; !sampleSizeDefined {
 		ingestionProps["sample_size"] = 200000
 	}
+	// set union_by_name to unify the schema of the files
+	if _, defined := ingestionProps["union_by_name"]; !defined {
+		ingestionProps["union_by_name"] = true
+	}
```

Comment:
If the user explicitly set `union_by_name` to false, we should check that allow column addition is false